### PR TITLE
[master < T1094] Extend insert on record to accept mgp::Value

### PR DESCRIPTION
**Pull request by mpintaric55334**
This is an additional overload of Record::Insert function. While writing collections query module in  C++ API, the need for returning mgp::Any datatype in AddProcedure was discovered. The changes in this PR adress this problem, and by overloading Record::Insert to accept mgp::Value, mgp::Any datatype can be inserted into Record.

[master < Task] PR
- [ ] Check, and update documentation if necessary
- [x] Provide the full content or a guide for the final git message

To keep docs changelog up to date, one more thing to do:
- [X] Write a release note here 
You are now able to return mgp::Type::Any inside Record when writing query modules in C++ API.

- [X] Tag someone from docs team in the comments

### CODE REVIEW

**Review by antoniofilipovic — APPROVED** [review]
I think this should be fine :smile: @Josipmrden please check this one and check if we can test it somehow

File: include/mgp.hpp [hunk 1]
```diff
@@ -3276,6 +3278,53 @@ inline void Record::Insert(const char *field_name, const Duration &duration) {
   mgp::value_destroy(mgp_val);
 }
 
+inline void Record::Insert(const char *field_name, const Value &value) {
+  switch (value.Type()) {
+    case Type::Bool:
+      Insert(field_name, value.ValueBool());
+      break;
```
Comment (antoniofilipovic) [inline include/mgp.hpp:3285]: Actually you can combine these two into return Insert, so please change that I gave approve too soon :joy:

**Review by antoniofilipovic — CHANGES_REQUESTED** [review]
Actually needs change, small change

File: include/mgp.hpp (hunk 1 above)
Comment (mpintaric55334) [reply include/mgp.hpp]: Done

**Review by antoniofilipovic — APPROVED** [review]
Good work!

**Comment by Josipmrden** [comment]
@mpintaric55334 Please edit the PR message, and add an appropriate message why was this added (reasoning.)

**Comment by mpintaric55334** [comment]
Implemented Record::Insert for mgp::Value, because it is necessary if you set mgp::Any as return type in AddProcedure

**Comment by Josipmrden** [comment]
https://github.com/memgraph/memgraph/pull/982 @mpintaric55334  here is an example of how it should look like, You should only leave the master < task with the checkboxes and tick all that you have done + write a release note message so @vpavicic can reason why was this added

**Comment by mpintaric55334** [comment]
@vpavicic I will start writing documentation when I get the green light and instructions how to do that from Antonio or Josip.

**Comment by vpavicic** [comment]
> @vpavicic I will start writing documentation when I get the green light and instructions how to do that from Antonio or Josip.

OK... I can give you instructions how to write the docs as well, but whatever you prefer :)

**Review by Josipmrden — APPROVED** [review]

**Pull request merged**